model-client/src/jvmMain/kotlin/org/modelix/model/client2/LazyLoading.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2024.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.model.client2
+
+import kotlinx.coroutines.runBlocking
+import org.modelix.model.IKeyListener
+import org.modelix.model.IKeyValueStore
+import org.modelix.model.IVersion
+import org.modelix.model.lazy.BranchReference
+import org.modelix.model.lazy.CLVersion
+import org.modelix.model.lazy.CacheConfiguration
+import org.modelix.model.lazy.ObjectStoreCache
+import org.modelix.model.lazy.RepositoryId
+import org.modelix.model.persistent.HashUtil
+
+/**
+ * IModelClientV2#loadVersion eagerly loads the whole model. For large models this can be slow and requires lots of
+ * memory.
+ * This function loads parts of the model lazily while it is iterated and limits the amount of data that is cached on
+ * the client side.
+ * To reduce the relative overhead of requests to the server, the lazy loading algorithm tries to predict which nodes
+ * are required next and fill a "prefetch cache" by using "free capacity" of the regular requests. That means,
+ * the number of requests doesn't change by this prefetching, but small requests are filled to up to their limit with
+ * additional prefetch requests.
+ */
+fun IModelClientV2.lazyLoadVersion(repositoryId: RepositoryId, versionHash: String, config: CacheConfiguration): IVersion {
```

Comment:
Lacks some documentation on what is a `CacheConfiguration` and where to acquire it.

Reply:
I added documentation to the values in `CacheConfiguration`. I will not document how to create an instance of a class.

Reply:
What would happen here if I put a completely different cache configuration in subsequent calls to this method? Would an already existing cache with the config I provided first be adapted?

Reply:
There is no shared cache. Each call to this method creates its own cache.